app/controllers/admin_api/books_controller.rb

```diff
@@ -3,9 +3,15 @@ class AdminApi::BooksController < ApplicationController
   before_action :set_book, only: [:edit, :update, :destroy, :show]
 
   def index
-    @books = Book.search(params[:search])
-                 .order_by(params[:sort_field], params[:sort_order])
-                 .page(params[:page])
+    if params[:sort_field] == "Most popular"
+      @books = Kaminari.paginate_array(Book.search(params[:search]).most_popular)
+                       .page(params[:page])
+    else
+      @books = Book.search(params[:search])
+                   .order_by(params[:sort_field], params[:sort_order])
+                   .page(params[:page])
+                   .decorate
+    end
     @sortable_fields = Book.sort_options
   end
```

Comment:
So what's `index` actually doing now? It looks to be setting `@sortable_fields`, `@books`, and organizing books. That's a lot. Can you give me the rundown?